dataland-frontend/tests/e2e/fixtures/sme/SmeDataFixtures.ts

```diff
@@ -1,103 +1,112 @@
       sites: {
-        listOfProductionSites: valueOrUndefined(generateArray(generateProductionSite), undefinedProbability),
+        listOfProductionSites: dataGenerator.generateProductionSite(),
       },
       products: {
-        listOfProducts: valueOrUndefined(generateArray(generateProduct), undefinedProbability),
+        listOfProducts: dataGenerator.generateProduct(),
       },
     },
     power: {
       investments: {
-        percentageOfInvestmentsInEnhancingEnergyEfficiency: valueOrUndefined(
-          getRandomPercentageRangeInvestmentEnergyEfficiency(),
+        percentageOfInvestmentsInEnhancingEnergyEfficiency: dataGenerator.valueOrUndefined(
+          generatePercentageRangeInvestmentEnergyEfficiency(),
         ),
       },
       consumption: {
-        powerConsumptionInMwh: valueOrUndefined(randomInt(2000), undefinedProbability),
-        powerFromRenewableSources: valueOrUndefined(randomYesNo(), undefinedProbability),
-        energyConsumptionHeatingAndHotWater: valueOrUndefined(randomInt(1000), undefinedProbability),
-        primaryEnergySourceForHeatingAndHotWater: valueOrUndefined(getRandomHeatSource()),
-        energyConsumptionCoveredByOwnRenewablePowerGeneration: valueOrUndefined(
-          getRandomPercentageRangeEnergyConsumption(),
+        powerConsumptionInMwh: dataGenerator.valueOrUndefined(generateInt(2000)),
+        powerFromRenewableSources: dataGenerator.randomYesNo(),
+        energyConsumptionHeatingAndHotWater: dataGenerator.randomInt(1000),
+        primaryEnergySourceForHeatingAndHotWater: dataGenerator.valueOrUndefined(generateHeatSource()),
+        energyConsumptionCoveredByOwnRenewablePowerGeneration: dataGenerator.valueOrUndefined(
+          generatePercentageRangeEnergyConsumption(),
         ),
       },
     },
     insurances: {
       naturalHazards: {
-        insuranceAgainstNaturalHazards: valueOrUndefined(randomYesNo(), undefinedProbability),
-        amountCoveredByInsuranceAgainstNaturalHazards: valueOrUndefined(randomInt(50000000), undefinedProbability),
-        naturalHazardsCovered: valueOrUndefined(getRandomSlectionOfNaturalHazards(), undefinedProbability),
+        insuranceAgainstNaturalHazards: dataGenerator.randomYesNo(),
+        amountCoveredByInsuranceAgainstNaturalHazards: dataGenerator.randomInt(50000000),
+        naturalHazardsCovered: dataGenerator.valueOrUndefined(generateSelectionOfNaturalHazards()),
       },
     },
   };
 }
+
+class SmeGenerator extends Generator {
+  /**
+   * Generates a random product
+   * @returns a random product
+   */
+  generateProduct(): SmeProduct[] | undefined {
+    return this.randomArray(() => {
+      return {
+        name: faker.commerce.productName(),
+        percentageOfTotalRevenue: this.randomPercentageValue(),
+      } as SmeProduct;
+    });
+  }
+
+  /**
+   * Generates a random production site
+   * @returns a random production site
+   */
+  generateProductionSite(): SmeProductionSite[] | undefined {
```

Comment:
Same comment as for "generateProduct()" => If it can return "undefined", it should be named "randomProductionSites()"

Reply:
renamed